ui/pages/send/send-content/send-hex-data-row/send-hex-data-row.component.js

```diff
@@ -13,16 +13,36 @@ export default class SendHexDataRow extends Component {
     t: PropTypes.func,
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      hexValue: null,
+    };
+  }
+
   onInput = (event) => {
     const { updateSendHexData } = this.props;
     const data = event.target.value.replace(/\n/gu, '') || null;
     updateSendHexData(data);
   };
 
+  componentDidMount() {
+    this.setState({ hexValue: this.props.data });
+  }
+
+  componentDidUpdate(prevProps) {
+    const { data: prevPropsHexValue } = prevProps;
+    const { data: propsHexValue } = this.props;
+
+    if (prevPropsHexValue !== propsHexValue) {
+      this.setState({ hexValue: propsHexValue });
+    }
```

Comment:
In react , `props` are indeed intended to be read-only. That’s not possible to change the value of `props` directly within a react component . If you need to modify a value based on the `props`, you should create and manage `state` within the component.

the error we've had is due to the fact that we're trying to change the value of the `props` directly, so as long as we don't refresh we're always left with the value of the previous `props` which doesn't change.


the solution was to manage this through states within the component